lib/nodejs/persistence.js

```diff
@@ -249,21 +249,26 @@ function HandlePersistenceSave( request, response, parsedRequest, segments ) {
         CreateSaveDirectory( parsedRequest[ 'public_path' ], saveName, saveRevision );
         var persistenceData = "";
         var persistenceLength = 0;
-        request.on( 'data', function ( data ) {
-            persistenceData += data;
-            if ( persistenceData.length > 50000000 ) {
-                persistenceData = "";
-                response.writeHead( 413, { 'Content-Type': 'text/plain' } ).end( );
-                request.connection.destroy();
-            }
-        } );
-        request.on( 'end', function( ) {
-            if ( persistenceData.length > 0 ) {
-                var processedPost = querystring.parse( persistenceData );
-                fs.writeFileSync( helpers.JoinPath( './documents', parsedRequest[ 'public_path' ], saveName, 'saveState' + processedPost[ "extension" ] ), processedPost[ "jsonState" ] );
-                fs.writeFileSync( helpers.JoinPath( './documents', parsedRequest[ 'public_path' ], saveName, 'saveState_' + saveRevision + processedPost[ "extension" ] ), processedPost[ "jsonState" ] );
-            }
-        } );
+        if ( typeof request.body === "object" ) {
+            fs.writeFileSync( helpers.JoinPath( './documents', parsedRequest[ 'public_path' ], saveName, 'saveState' + request.body[ "extension" ] ), request.body[ "jsonState" ] );
+            fs.writeFileSync( helpers.JoinPath( './documents', parsedRequest[ 'public_path' ], saveName, 'saveState_' + saveRevision + request.body[ "extension" ] ), request.body[ "jsonState" ] );
+        } else {
+            request.on( 'data', function ( data ) {
+                persistenceData += data;
+                if ( persistenceData.length > 50000000 ) {
```

Comment:
Do I understand correctly that this is guarding to make sure that someone doesn't try to save something that is too large and overwhelms the server disk space?  And that if it is too large it will silently fail?